File: client/types/task.ts
export interface Task {
  task_id: number;
  task_title: string;
  group_id: number;
  created_by: string;
  start_date: string | null;
  end_date?: string | null;
  quick_task: boolean;
  notes: string;
  repeating: boolean;
  repeating_interval?: string | null;
  repeating_end_date?: string | null;
  task_status: string;
  task_type: string;
  task_info?: string | null;
  label: string;
  assigned_to: string;
  [key: string]: string | number | boolean | null | undefined; // Index signature for string indexing
}

File: backend/models/task.go
package models

import (
	"time"
)

type Task struct {
	TaskID            int        `json:"task_id"`
	TaskTitle         string     `json:"task_title"`
	GroupID           int        `json:"group_id"`
	CreatedBy         string     `json:"created_by"` // User ID
	CreatedDate       time.Time  `json:"created_date"`
	StartDate         *time.Time `json:"start_date"`
	EndDate           *time.Time `json:"end_date"`
	QuickTask         bool       `json:"quick_task"`
	Notes             *string    `json:"notes"`
	Repeating         bool       `json:"repeating"`
	RepeatingInterval *string    `json:"repeating_interval"`
	RepeatingEndDate  *time.Time `json:"repeating_end_date"`
	TaskStatus        string     `json:"task_status"`
	TaskType          string     `json:"task_type"`
	TaskInfo          *string    `json:"task_info"`
}
